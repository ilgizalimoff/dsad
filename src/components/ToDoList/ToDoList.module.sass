.wrapper
    display: flex
    justify-content: center

.todo
   min-width: 50%

.addBtn
    width: 100%
    background: transparent
    outline: none
    border: 1px solid gray
    padding: 10px
    border-radius: 10px
    font-size: 16px
    margin-bottom: 20px
    cursor: pointer

.selectBlock
    select
        width: 100%
        border: 1px solid gray
        padding: 10px
        border-radius: 10px
        font-size: 16px
        outline: none
        cursor: pointer
        text-align: center

.empty
    font-weight: 400
    text-align: center
    margin-top: 20px