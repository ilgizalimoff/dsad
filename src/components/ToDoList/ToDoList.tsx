import { useState, useEffect } from 'react';
import styles from './ToDoList.module.sass';
import ToDoItem from '../ToDoItem/ToDoItem';
import Modal from '../Modal/Modal';
import { getTodosBySort } from '../../api/ToDoService';
import { options } from '../../const/const';
import { useAppSelector } from '../../hook';

type IOption = {
    title: string
    query: string | boolean
}

const ToDoList: React.FC = () => {
    const todos = useAppSelector(state => state.todos.list);

    const [visible, setVisible] = useState(false)

    const [sort, setSort] = useState<string | boolean>('all')

    const onSelectChange = async (sortMethod: string | boolean) => {
        // let currentSort = options.filter(option => option.title === sortMethod)[0].query
        // setSort(currentSort)
        // await getTodosBySort(currentSort).then(data => todo.setTodos(data))
    }

    useEffect(() => {
        // getTodosBySort(sort).then(data => todo.setTodos(data))
    }, [])

    return (
        <div className={styles.wrapper}>
            <div className={styles.todo}>
                <button
                    data-testid='addTodoBtn'
                    className={styles.addBtn}
                    onClick={() => setVisible(true)}>
                    Добавить в список
                </button>

                <Modal
                    visible={visible}
                    setVisible={setVisible} />

                <div className={styles.selectBlock}>
                    <select onChange={(e) => onSelectChange(e.target.value)}>
                        {options.map((option: IOption) =>
                            <option key={option.title}>
                                {option.title}
                            </option>
                        )}
                    </select>
                </div>

                {todos.length === 0
                    ? <h3 className={styles.empty}>Список пуст...</h3>
                    : todos.map(el =>
                        <ToDoItem key={el.id} todoItem={el} sort={sort} />
                    )
                }
            </div>
        </div>
    );
}

export default ToDoList;