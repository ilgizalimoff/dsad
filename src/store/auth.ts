import { makeAutoObservable } from "mobx"

class ToDo {
    _user = [
        {
            "id": 1,
            "lgoin": "",
            "password": ""
        }
    ]

    _isAuth = false

    constructor() {
        makeAutoObservable(this)
    }

    setUser(todos: any) {
        this._user = todos
    }

    setIsAuth(bool: boolean) {
        this._isAuth = bool
    }

    get user() {
        return this._user
    }
    
    get isAuth() {
        return this._isAuth
    }
}

export default new ToDo()