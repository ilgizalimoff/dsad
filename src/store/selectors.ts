import { IToDoItem } from "../const/const";

type TodosState = {
    todos: IToDoItem[];
}

export const selectTodos = (state: TodosState) => state.todos