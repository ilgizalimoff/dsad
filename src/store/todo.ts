import { makeAutoObservable } from "mobx"
import { IToDoItem } from "../const/const"

class ToDo {
    _todos = [
        {
            "id": 1,
            "body": "",
            "complete": false
        }
    ]

    constructor() {
        makeAutoObservable(this)
    }

    setTodos(todos: IToDoItem[]) {
        this._todos = todos
    }

    get fetchToDos() {
        return this._todos
    }
}

export default new ToDo()