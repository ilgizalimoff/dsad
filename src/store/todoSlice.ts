import { createSlice, PayloadAction } from '@reduxjs/toolkit';
import { IToDoItem } from '../const/const';

type TodosState = {
  list: IToDoItem[];
}

const initialState: TodosState = {
  list: [],
}

const todoSlice = createSlice({
  name: 'todos',
  initialState,
  reducers: {
    addTodo(state, action: PayloadAction<string>) {
      state.list.push({
        id: Date.now(),
        body: action.payload,
        complete: false,
      });
    },

    toggleComplete(state, action: PayloadAction<number>) {
      const toggledTodo = state.list.find(todo => todo.id === action.payload);
      if (toggledTodo) {
        toggledTodo.complete = !toggledTodo.complete;
      }
    },

    removeTodo(state, action: PayloadAction<number>) {
      state.list = state.list.filter(todo => todo.id !== action.payload);
    }
  },
});

export const { addTodo, toggleComplete, removeTodo } = todoSlice.actions;

export default todoSlice.reducer;