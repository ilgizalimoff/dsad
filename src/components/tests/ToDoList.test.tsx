import { fireEvent, render, screen } from "@testing-library/react"
import '@testing-library/jest-dom/extend-expect'; // нужно добавить эту строку
import ToDoList from "../ToDoList/ToDoList"
import * as reduxHooks from 'react-redux'
import * as actions from '../../store/todoSlice'

jest.mock('react-redux')

const mockedUseSelector = jest.spyOn(reduxHooks, 'useSelector')

const mockedDispatch = jest.spyOn(reduxHooks, 'useDispatch')

const todos = [
    {
        "id": 1683893881802,
        "body": "Сходить в магазин",
        "complete": false
    },
    {
        "id": 1683893901661,
        "body": "Помыть посуду",
        "complete": false
    }
]

describe('Проверка компонента TodoList', () => {
    it(('Проверка рендера компонента TodoList с пустым массивом данных'), () => {
        mockedUseSelector.mockReturnValue([])
        const component = render(<ToDoList />)
        expect(component).toMatchSnapshot()
    })

    it(('Проверка рендера компонента TodoList конкретным массивом данных'), () => {
        mockedUseSelector.mockReturnValue(todos)
        const component = render(<ToDoList />)
        expect(component).toMatchSnapshot()
    })

    it('Проверка рендера компонента Modal', () => {
        mockedUseSelector.mockReturnValue(todos)
        render(<ToDoList />)
        fireEvent.click(screen.getByTestId('addTodoBtn'))
        expect(screen.queryByTestId('modal')).toBeInTheDocument()
    })
})