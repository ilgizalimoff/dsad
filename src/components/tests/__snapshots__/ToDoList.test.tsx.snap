// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`todoSlice should create TodoList with empty todos 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="wrapper"
      >
        <div
          class="todo"
        >
          <button
            class="addBtn"
            data-testid="addTodoBtn"
          >
            Добавить в список
          </button>
          <div
            class="selectBlock"
          >
            <select>
              <option>
                Все
              </option>
              <option>
                Выполненные
              </option>
              <option>
                Невыполненные
              </option>
            </select>
          </div>
          <h3
            class="empty"
          >
            Список пуст...
          </h3>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="wrapper"
    >
      <div
        class="todo"
      >
        <button
          class="addBtn"
          data-testid="addTodoBtn"
        >
          Добавить в список
        </button>
        <div
          class="selectBlock"
        >
          <select>
            <option>
              Все
            </option>
            <option>
              Выполненные
            </option>
            <option>
              Невыполненные
            </option>
          </select>
        </div>
        <h3
          class="empty"
        >
          Список пуст...
        </h3>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`todoSlice should create TodoList with todo items 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="wrapper"
      >
        <div
          class="todo"
        >
          <button
            class="addBtn"
            data-testid="addTodoBtn"
          >
            Добавить в список
          </button>
          <div
            class="selectBlock"
          >
            <select>
              <option>
                Все
              </option>
              <option>
                Выполненные
              </option>
              <option>
                Невыполненные
              </option>
            </select>
          </div>
          <div
            class="item"
            data-testid="todoItem"
          >
            <div>
              <input
                data-testid="checkboxBtn"
                type="checkbox"
              />
              <h3>
                Сходить в магазин
              </h3>
            </div>
            <button
              data-testid="removeBtn"
            >
              Удалить
            </button>
          </div>
          <div
            class="item"
            data-testid="todoItem"
          >
            <div>
              <input
                data-testid="checkboxBtn"
                type="checkbox"
              />
              <h3>
                Помыть посуду
              </h3>
            </div>
            <button
              data-testid="removeBtn"
            >
              Удалить
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="wrapper"
    >
      <div
        class="todo"
      >
        <button
          class="addBtn"
          data-testid="addTodoBtn"
        >
          Добавить в список
        </button>
        <div
          class="selectBlock"
        >
          <select>
            <option>
              Все
            </option>
            <option>
              Выполненные
            </option>
            <option>
              Невыполненные
            </option>
          </select>
        </div>
        <div
          class="item"
          data-testid="todoItem"
        >
          <div>
            <input
              data-testid="checkboxBtn"
              type="checkbox"
            />
            <h3>
              Сходить в магазин
            </h3>
          </div>
          <button
            data-testid="removeBtn"
          >
            Удалить
          </button>
        </div>
        <div
          class="item"
          data-testid="todoItem"
        >
          <div>
            <input
              data-testid="checkboxBtn"
              type="checkbox"
            />
            <h3>
              Помыть посуду
            </h3>
          </div>
          <button
            data-testid="removeBtn"
          >
            Удалить
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Проверка компонента TodoList Проверка рендера компонента TodoList конкретным массивом данных 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="wrapper"
      >
        <div
          class="todo"
        >
          <button
            class="addBtn"
            data-testid="addTodoBtn"
          >
            Добавить в список
          </button>
          <div
            class="selectBlock"
          >
            <select>
              <option>
                Все
              </option>
              <option>
                Выполненные
              </option>
              <option>
                Невыполненные
              </option>
            </select>
          </div>
          <div
            class="item"
            data-testid="todoItem"
          >
            <div>
              <input
                data-testid="checkboxBtn"
                type="checkbox"
              />
              <h3>
                Сходить в магазин
              </h3>
            </div>
            <button
              data-testid="removeBtn"
            >
              Удалить
            </button>
          </div>
          <div
            class="item"
            data-testid="todoItem"
          >
            <div>
              <input
                data-testid="checkboxBtn"
                type="checkbox"
              />
              <h3>
                Помыть посуду
              </h3>
            </div>
            <button
              data-testid="removeBtn"
            >
              Удалить
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="wrapper"
    >
      <div
        class="todo"
      >
        <button
          class="addBtn"
          data-testid="addTodoBtn"
        >
          Добавить в список
        </button>
        <div
          class="selectBlock"
        >
          <select>
            <option>
              Все
            </option>
            <option>
              Выполненные
            </option>
            <option>
              Невыполненные
            </option>
          </select>
        </div>
        <div
          class="item"
          data-testid="todoItem"
        >
          <div>
            <input
              data-testid="checkboxBtn"
              type="checkbox"
            />
            <h3>
              Сходить в магазин
            </h3>
          </div>
          <button
            data-testid="removeBtn"
          >
            Удалить
          </button>
        </div>
        <div
          class="item"
          data-testid="todoItem"
        >
          <div>
            <input
              data-testid="checkboxBtn"
              type="checkbox"
            />
            <h3>
              Помыть посуду
            </h3>
          </div>
          <button
            data-testid="removeBtn"
          >
            Удалить
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Проверка компонента TodoList Проверка рендера компонента TodoList с пустым массивом данных 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="wrapper"
      >
        <div
          class="todo"
        >
          <button
            class="addBtn"
            data-testid="addTodoBtn"
          >
            Добавить в список
          </button>
          <div
            class="selectBlock"
          >
            <select>
              <option>
                Все
              </option>
              <option>
                Выполненные
              </option>
              <option>
                Невыполненные
              </option>
            </select>
          </div>
          <h3
            class="empty"
          >
            Список пуст...
          </h3>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="wrapper"
    >
      <div
        class="todo"
      >
        <button
          class="addBtn"
          data-testid="addTodoBtn"
        >
          Добавить в список
        </button>
        <div
          class="selectBlock"
        >
          <select>
            <option>
              Все
            </option>
            <option>
              Выполненные
            </option>
            <option>
              Невыполненные
            </option>
          </select>
        </div>
        <h3
          class="empty"
        >
          Список пуст...
        </h3>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
